private/src/avb_streamhandler/IasAvbReceiveEngine.cpp

```diff
@@ -964,6 +995,16 @@ IasResult IasAvbReceiveEngine::run()
         (void) dispatchPacket(it->second, NULL, 0u, now);
       }
       (void) unlock();
+
+	if (mWatchdog) {
```

Comment:
Hint: maybe comment why each reset/register/unregister of the watchdog takes place on different places of method `run()`, just like in the transmitter? This way future people can know why the watchdog control is necessary in each place, so code can be properly changed.

Reply:
Yes, I added these based on the way they did it for the transmit thread. Will explain them.